src/ReportManager/SchemaFormContext/SchemaForm/index.test.js

```diff
@@ -0,0 +1,155 @@
+        'header-ghqdjqGinaJMptIEJBQmO': {
+          'label': 'A great header',
+          'section': 'section-_PdgePvPWyACfu9sgN_F6',
+          'size': 'LARGE'
+        }
+      },
+      'order': [
+        'section-_PdgePvPWyACfu9sgN_F6'
+      ],
+      'sections': {
+        'section-_PdgePvPWyACfu9sgN_F6': {
+          'columns': 1,
+          'isActive': true,
+          'label': '',
+          'leftColumn': [
+            {
+              'name': 'header-ghqdjqGinaJMptIEJBQmO',
+              'type': 'header'
+            },
+            {
+              'name': 'this_is_a_text',
+              'type': 'field'
+            }
+          ],
+          'rightColumn': []
+        }
+      }
+    }
+  };
+
+  const initialProps = {
+    schema: efb,
+    formData: {
+      'this_is_a_text': 'a text value'
+    },
+    onFormSubmit: () => {},
+    className: '',
+    onChange: () => {}
+  };
+
+  const renderSchemaForm = (props = initialProps) => {
+    return render(
+      <SchemaForm {...props} />
+    );
+  };
+
+  test('renders text field properly', () => {
+    renderSchemaForm();
+
+    const textFieldElement = screen.getByTestId('schema-form-text-field-this_is_a_text');
+
+    expect(within(textFieldElement).getByLabelText('This is a text')).toBeInTheDocument();
+    expect(within(textFieldElement).getByText('some good description')).toBeInTheDocument();
+
+    const inputField = within(textFieldElement).getByTestId('schema-form-text-field-input-this_is_a_text');
+    expect(inputField.value).toBe('a text value');
+    expect(inputField.placeholder).toBe('a placeholder');
+  });
+
+  test('update text field when user types in it', async () => {
+    const onChange = jest.fn();
+    renderSchemaForm({ ...initialProps, onChange });
+
+    const inputField = screen.getByTestId('schema-form-text-field-input-this_is_a_text');
+    await userEvent.type(inputField, 'A');
+
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(onChange).toHaveBeenCalledWith({
+      formData: {
+        'this_is_a_text': 'a text valueA'
+      }
+    });
+  });
+
+  test('show validation error for text field on submit', async () => {
+
+    const onSubmit = () => console.log('form submit');
+    const schema = { ...efb };
+    schema.json.properties.this_is_a_text.default = '';
+
+    renderSchemaForm({
+      ...initialProps,
+      schema,
+      onSubmit,
+      formData: {
+        'this_is_a_text': ''
+      }
+    });
+
+    const inputField = screen.getByTestId('schema-form-text-field-input-this_is_a_text');
+    await userEvent.type(inputField, '{enter}');
+
+    expect( screen.getByText('Error message') ).toBeVisible();
+  });
+
+  test('submit form with text field data', async () => {
+    const onFormSubmit = jest.fn();
+
+    renderSchemaForm({
+      ...initialProps,
+      onFormSubmit,
+    });
+
+    const inputField = screen.getByTestId('schema-form-text-field-input-this_is_a_text');
+
+    await userEvent.type(inputField, '{enter}');
```

Comment:
Do we want to do a form submission on enter?

Reply:
Currently is behaving like that in the old schemas for an event in the web app